apis/apps/v1alpha1/cluster_types.go

```diff
@@ -442,19 +442,37 @@ type InstanceTemplate struct {
 	// - It is the user's responsibility to ensure the node is suitable for the Pod.
 	// - If the node is unavailable, the Pod will remain in "Pending" state until the node is available or the Pod is deleted.
 	//
+	// Deprecated since v0.10, replaced by the `schedulingPolicy` field.
+	//
+	// +kubebuilder:deprecatedversion:warning="This field has been deprecated since 0.10.0"
+	// +kubebuilder:pruning:PreserveUnknownFields
 	// +optional
 	NodeName *string `json:"nodeName,omitempty"`
 
 	// Defines NodeSelector to override.
+	//
+	// Deprecated since v0.10, replaced by the `schedulingPolicy` field.
+	//
+	// +kubebuilder:deprecatedversion:warning="This field has been deprecated since 0.10.0"
+	// +kubebuilder:pruning:PreserveUnknownFields
 	// +optional
 	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
 
 	// Tolerations specifies a list of tolerations to be applied to the Pod, allowing it to tolerate node taints.
 	// This field can be used to add new tolerations or override existing ones.
 	//
+	// Deprecated since v0.10, replaced by the `schedulingPolicy` field.
+	//
+	// +kubebuilder:deprecatedversion:warning="This field has been deprecated since 0.10.0"
+	// +kubebuilder:pruning:PreserveUnknownFields
 	// +optional
 	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
```

Comment:
should remove them directly as they are introduced in the 0.9

Reply:
got it